modules/certificates/main.tf

```diff
@@ -0,0 +1,104 @@
+resource "kubernetes_namespace" "cert_manager" {
+  count = var.install_cert_manager ? 1 : 0
+
+  metadata {
+    name = var.cert_manager_namespace
+  }
+}
+
+resource "helm_release" "cert_manager" {
+  count = var.install_cert_manager ? 1 : 0
+
+  # cert-manager is a singleton resource for the cluster,
+  # so not using name prefixes here.
+  name       = "cert-manager"
+  namespace  = kubernetes_namespace.cert_manager[0].metadata[0].name
+  repository = "https://charts.jetstack.io"
+  chart      = "cert-manager"
+  version    = "v1.17.1"
+
+  set {
+    name  = "crds.enabled"
+    value = "true"
+  }
+
+  depends_on = [
+    kubernetes_namespace.cert_manager,
+  ]
+}
+
+resource "kubernetes_manifest" "self_signed_cluster_issuer" {
+  # only create these after cert manager is installed
+  count = var.use_self_signed_cluster_issuer ? 1 : 0
+
+  manifest = {
+    "apiVersion" = "cert-manager.io/v1"
+    "kind"       = "ClusterIssuer"
+    "metadata" = {
+      "name" = "${var.name_prefix}-self-signed"
+    }
+    "spec" = {
+      "selfSigned" = {}
+    }
+  }
+
+  depends_on = [
+    helm_release.cert_manager,
+  ]
+}
+
+resource "kubernetes_manifest" "self_signed_root_ca_certificate" {
+  count = var.use_self_signed_cluster_issuer ? 1 : 0
+
+  manifest = {
+    "apiVersion" = "cert-manager.io/v1"
+    "kind"       = "Certificate"
+    "metadata" = {
+      "name"      = "${var.name_prefix}-self-signed-ca"
+      "namespace" = var.cert_manager_namespace
+    }
+    "spec" = {
+      "isCA"       = true
+      "commonName" = "${var.name_prefix}-self-signed-ca"
+      "secretName" = "${var.name_prefix}-root-ca"
+      "privateKey" = {
+        "algorithm"      = "RSA"
+        "encoding"       = "PKCS8"
+        "size"           = 4096
+        "rotationPolicy" = "Always"
```

Comment:
Do we want to abstract those as vars or are we ok with the defaults here in every case?

Reply:
I guess we could move them to a var, but I think if they want them changed, they probably want to define their own certificates. There are many options on the Certificate that aren't configured here, and making them variables would likely get more unwieldy than just leaving it all to the user.

I plan to change `terraform-helm-materialize` to make the namespace creation optional, and then it will be easier for them to define their own Issuers. If they don't like our defaults for ClusterIssuer, they can already define one of their own.